src/coord/src/coord.rs

```diff
@@ -976,6 +940,58 @@ where
         );
     }
 
+    /// Inserts a single row into the MZ_DATABASES catalog view, with diff `diff`.
+    ///
+    /// # Arguments
+    ///
+    /// * `global_id`:   Global id of the database.
+    /// * `name`:        Name of the database.
+    /// * `diff`:        The number of updates to perform. A positive number indicates an addition
+    ///                  of the row, a negative number indicates a subtraction.
+    fn update_mz_databases_catalog_view(&mut self, global_id: i64, name: &str, diff: isize) {
+        self.update_catalog_view(
+            MZ_DATABASES.id,
+            iter::once((
+                Row::pack(&[Datum::String(&global_id.to_string()), Datum::String(&name)]),
+                diff,
+            )),
+        )
+    }
+
+    /// Inserts a single row into the MZ_SCHEMAS catalog view, with diff `diff`.
+    ///
+    /// # Arguments
+    ///
+    /// * `database_id`: String representation of the database in which the schema is present.
+    ///                  Possible values are an actual database's global id, or "AMBIENT".
+    /// * `schema_id`:   Id of the schema.
+    /// * `schema_name`: Name of the schema.
+    /// * `typ`:         There are two types of schemas: "SYSTEM" and "USER". This classification
+    ///                  impacts which schemas can be updated or deleted.
+    /// * `diff`:        The number of updates to perform. A positive number indicates an addition
+    ///                  of the row, a negative number indicates a subtraction.
```

Comment:
FWIW this seems like over-documentation for a method. Having documentation about the fields of the system views is 1000% great, but e.g. explaining the "system"/"user" classification here is too much detail IMO for a method that updates that view. Personally I would write 

```rust
/// Inserts or removes a row from [`builtin::MZ_SCHEMAS`] based on the supplied `diff`.
```

(Actually, to be totally honest, I probably wouldn't bother documenting it all, since it's private API that's well named and very straightforward, but I know that philosophy makes some people anxious.)

What would be _extremely_ high value is if you could move these great descriptions of the fields of each catalog view into user-facing docs on materialize.io/docs. @sploiselle might have thoughts about how to structure the docs  slash where in the docs hierarchy they should live.

If you want this level of detail somewhere in the codebase, I think it could make sense to _also_ put these descriptions on the structs themselves in builtin.rs, though again, personally, I'd just slap a module-level comment in builtin.rs that points developers to the user docs explaining the fields, to avoid having to put the docs in two places.

Reply:
Good point! I will fix that [here](https://github.com/MaterializeInc/materialize/pull/4085) too.

Reply:
Yeah, very glad to figure out where to put this in the docs, esp. since you've already done the hard part (i.e. writing and formatting). I can't quite figure out where this data gets exposed in userland, but glad to chat.